src/vivarium/framework/results/manager.py

```diff
@@ -75,29 +75,43 @@ def setup(self, builder: "Builder"):
 
     def on_post_setup(self, _: Event):
         """Initialize self._metrics with 0s DataFrame' for each measure and all stratifications"""
+        registered_stratifications = self._results_context.stratifications
+        registered_stratification_names = set(
+            stratification.name for stratification in registered_stratifications
+        )
+
+        all_missing_stratifications = {}
+        unused_stratifications = registered_stratification_names.copy()
+
         for event_name in self._results_context.observations:
             for (
                 _pop_filter,
-                stratification_names,
+                all_requested_stratification_names,
             ), observations in self._results_context.observations[event_name].items():
                 for observation in observations:
                     measure = observation.name
-                    observation_stratifications = [
+
+                    # Batch missing stratifications
+                    missing_stratifications = set(
```

Comment:
Nit: you're performing `set(all_requested_stratification_names)` twice. You could save it as a local variable.